query/resource_selector.go

```diff
@@ -225,18 +226,41 @@ func SetResourceSelectorClause(ctx context.Context, resourceSelector types.Resou
 	}
 
 	if resourceSelector.Search != "" {
-		var prefixQueries []*gorm.DB
-		if resourceSelector.Name == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("name ILIKE ?", resourceSelector.Search+"%"))
-		}
-		if resourceSelector.TagSelector == "" && table == "config_items" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(tags) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
-		}
-		if resourceSelector.LabelSelector == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(labels) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+		if strings.HasPrefix(resourceSelector.Search, "component_config_traverse") && table == "components" {
+			// search: component_config_traverse=72143d48-da4a-477f-bac1-1e9decf188a6,outgoing,Kubernetes::Pod,Kubernetes::Node
```

Comment:
Can we use this parser to start with: https://github.com/flanksource/mission-control/blob/main/cmd/catalog.go#L27

`component_config_traverse=72143d48-da4a-477f-bac11e9decf188a6
 direction=outgoing type=Kubernetes::Pod,Kubernetes::Node`

Reply:
okay, updating